pip/pip-321.md

```diff
@@ -0,0 +1,251 @@
+Add a new endpoint `setNamespaceAllowedClusters` to the REST API.
+```
+Type: POST
+Path: /namespaces/{tenant}/{namespace}/allowedcluster
+Query parameters: tenant, namespace
+HTTP body parameters: List<String> clusterIds
+Response code:
+  * 400 Bad request, Allowed-clusters should contain all replication-clusters
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 409 Conflict, Peer-cluster can't be part of allowed-cluster.
+  * 412 Precondition Failed, Namespace is not global or invalid cluster ids.
+Explanation for 409 Conflict: This follows the behavior of namespace replication clusters. In the PIP-8, quoted in background knowledge, it introduce the concept of peer-cluster and peer-cluster can't be part of replication-cluster. For the allowed-clusters, users could enable replication at the topic level, so the peer-cluster can't be part of allowed-clusters too.
+```
+
+Add a new endpoint `getNamespaceAllowedClusters` to the REST API.
+```
+type: GET
+Path: /namespaces/{tenant}/{namespace}/allowedcluster
+Query parameters: tenant, namespace
+Response code:
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 412 Precondition Failed, Namespace is not global.
+```
+
+Add a new endpoint `setNamespaceTopicPolicySynchronizedClusters` to the REST API.
+```
+Type: POST
+Path: /namespaces/{tenant}/{namespace}/topicpolicycluster
+Query parameters: tenant, namespace
+HTTP body parameters: List<String> clusterIds
+Response code:
+  * 400 Bad request, Allowed-clusters should contain all topic-policy-synchronized-clusters
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 412 Precondition Failed, Namespace is not global or invalid cluster ids.
+```
+
+Add a new endpoint `getNamespaceTopicPolicySynchronizedClusters` to the REST API.
+```
+Type: GET
+Path: /namespaces/{tenant}/{namespace}/topicpolicycluster
+Query parameters: tenant, namespace
+Response code:
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 412 Precondition Failed, Namespace is not global.
+```
+
+### Binary protocol
+
+### Configuration
+
+### CLI
+
+### Metrics
+None.
+
+# Monitoring
+None.
+# Security Considerations
+If the broker enables authentication, then this configuration can only be set by the client who was authenticated.
+
+# Backward & Forward Compatibility
+
+## Revert
+This can not be reverted if they enable geo-replication at the topic level. Otherwise, the replicator of these topics will be closed.
+
+## Upgrade
+There are no additional operations that need to be performed. The replication-clusters will be the default value of allowed-clusters and topic-policy-synchronized-clusters.
+
+# Alternatives
+
+Add a configuration `enableReplicationInTopicLevel` to make the topic able to be loaded in all clusters.
```

Comment:
Could you please explain more about it? We can't understand it in one line.

Reply:
This is for the `Problem 2: Global Topic Policies Not Replicating as Expected`. Delete it from this proposal too. 
https://github.com/apache/pulsar/pull/21648#discussion_r1435921717